storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -136,9 +136,14 @@ private Store(
     this.justified_checkpoint = justified_checkpoint;
     this.best_justified_checkpoint = best_justified_checkpoint;
     this.blocks = blocks;
-    this.votes = new HashMap<>(votes);
+
+    this.votes = new VoteTracker[VOTE_TRACKER_SIZE];
+    votes.forEach(
+        (key, value) -> {
+          this.votes[key.intValue()] = value;
+        });
     this.highestVotedValidatorIndex =
-        this.votes.keySet().stream().max(Comparator.naturalOrder()).orElse(UInt64.ZERO);
+        votes.keySet().stream().max(Comparator.naturalOrder()).orElse(UInt64.ZERO);
```

Comment:
If we used a normal `for` loop here we could simplify find the max validator index at the same time as we populate the votes array/List.